src/EditorFeatures/CSharpTest/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -1403,8 +1383,89 @@ protected set
 }",
 @"class Class
 {
-
     public int P { get; protected set; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    public object Bar => bar;
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    public object Bar { get; } = new object();
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    public object Bar => bar; // prop comment
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    public object Bar { get; } = new object(); // prop comment
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    // doc
+    public object Bar => bar; // prop comment
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    // doc
+    public object Bar { get; } = new object(); // prop comment
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestKeepLeadingBlank()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+
+    private readonly object [|bar|] = new object();
```

Comment:
>bar [](start = 30, length = 3)

Would it make sense to test with some other fields or members above field `bar`?

Reply:
Have added tests.